numba/tests/test_linalg.py

```diff
@@ -720,5 +724,116 @@ def check(a, **kwargs):
                                   (np.array([[1., 2., ], [np.inf, np.nan]],
                                             dtype=np.float64),))
 
+
+class TestLinalgQr(TestLinalgBase):
+    """
+    Tests for np.linalg.qr.
+    """
+
+    def sample_matrix(self, size, dtype, order):
+        break_at = 10
+        jmp = 0
+        # have a few attempts at shuffling to get the condition number down
+        # else not worry about it
+        mn = size[0] * size[1]
+        np.random.seed(0)  # repeatable seed
+        while jmp < break_at:
```

Comment:
Well, its fine to return something, it's just that it may be poorly conditioned and as a result make comparison to numpy harder, which results in more tests going via reconstruction.